In [2]:
import pandas as pd 
import numpy as np

In [3]:
shopify = pd.read_excel(r'C:\Users\kchan\Desktop\New folder\shopify.xlsx')

In [5]:
print(shopify.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5000 entries, 0 to 4999
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   order_id        5000 non-null   int64         
 1   shop_id         5000 non-null   int64         
 2   user_id         5000 non-null   int64         
 3   order_amount    5000 non-null   int64         
 4   total_items     5000 non-null   int64         
 5   payment_method  5000 non-null   object        
 6   created_at      5000 non-null   datetime64[ns]
dtypes: datetime64[ns](1), int64(5), object(1)
memory usage: 273.6+ KB
None


In [7]:
print(shopify.total_items)

0       2
1       1
2       1
3       1
4       1
       ..
4995    2
4996    2
4997    3
4998    2
4999    2
Name: total_items, Length: 5000, dtype: int64


In [9]:

shopify['total_items'].value_counts()

2       1832
1       1830
3        941
4        293
5         77
2000      17
6          9
8          1
Name: total_items, dtype: int64

#####
We can see that there is a total of 17 orders that have an amount of 2000. This is the actual  reason why the average order value (AOV) is of such value (3145.13 CAD)

In [10]:
shopify[shopify['total_items'] == 2000]

order_id  shop_id  user_id  order_amount  total_items payment_method  \
15          16       42      607        704000         2000    credit_card   
60          61       42      607        704000         2000    credit_card   
520        521       42      607        704000         2000    credit_card   
1104      1105       42      607        704000         2000    credit_card   
1362      1363       42      607        704000         2000    credit_card   
1436      1437       42      607        704000         2000    credit_card   
1562      1563       42      607        704000         2000    credit_card   
1602      1603       42      607        704000         2000    credit_card   
2153      2154       42      607        704000         2000    credit_card   
2297      2298       42      607        704000         2000    credit_card   
2835      2836       42      607        704000         2000    credit_card   
2969      2970       42      607        704000         2000    credit_card   
3332      3333       42      607        704000         2000    credit_card   
4056      4057       42      607        704000         2000    credit_card   
4646      4647       42      607        704000         2000    credit_card   
4868      4869       42      607        704000         2000    credit_card   
4882      4883       42      607        704000         2000    credit_card   

              created_at  
15   2017-03-07 04:00:00  
60   2017-03-04 04:00:00  
520  2017-03-02 04:00:00  
1104 2017-03-24 04:00:00  
1362 2017-03-15 04:00:00  
1436 2017-03-11 04:00:00  
1562 2017-03-19 04:00:00  
1602 2017-03-17 04:00:00  
2153 2017-03-12 04:00:00  
2297 2017-03-07 04:00:00  
2835 2017-03-28 04:00:00  
2969 2017-03-28 04:00:00  
3332 2017-03-24 04:00:00  
4056 2017-03-28 04:00:00  
4646 2017-03-02 04:00:00  
4868 2017-03-22 04:00:00  
4882 2017-03-25 04:00:00

###### The fact of the matter here is that one individual with the user_id of 607 has bought a total of  34000 items which is drastically driving up the AOV due to the large skew of values as the next highest value with regards to total_items is 8.

In [13]:
## Mean(AOV) 
shopify[['order_amount']].mean()


order_amount    3145.128
dtype: float64

In [14]:
## Median (My Proposal Value)
shopify[['order_amount']].median()

order_amount    284.0
dtype: float64

## Question 1 Answers:

### Question 1a:

* The problem is that the AOV ( Average order value) is blindly calculated. Alongside that, the number of shoes that are being purchased within each order from each individual person at a specific time is not being accounted for just the order amount being divided by the amount of orders which doesn't take into account multiple orders.
* Secondly, another problem as specified is the 17 seperate orders of more than 2000 shoes. This is rapidly increasing the AOV value creating a very skewed analysis. 

### Question 1b:

* In my opinion, I believe that the best way to get a more robust or more accurate representation of the values we should use something that isnt affected by extreme values which in this case would be the mean.

### Question 1c:

* The median value as calculated above is 284. This is actually much more reasonable as this doesn't get affected by the extremev values and gives us a much nore robust and accurate analysis when going through the data.

## Question 2 Answers:

### Question 2a:

* First do the following query : SELECT ShipperID, count(*) FROM [Orders] group by ShipperID , this gives us the values and the shipper with shipperID 2 meaning the answer is:

### United Package with a value of 74.

### Question 2b:

* First do the following query : SELECT EmployeeID, count(*) FROM [Orders] group by EmployeeID , this will then groupby each EmployeeID and give us the counts, from this output we will see the highest value is for EmployeeID 4. 

* Next you want to run the following query : Select * FROM [Employees] WHERE EmployeeID = 4 , which will give us the individual with the Lastname:

### Peacock

### Question 2c:

* First do the following query : SELECT OrderID, count(*) FROM [Orders] Where CustomerID in (1,6,17,25,39,44,52,56,63,79,86) , this gives us the OrderID of 10267 which is the orders that each individual from ordered.
* Next, you want to run the following query : SELECT ProductID, count(*) FROM [OrderDetails] WHERE OrderID = 10267 which gives us the ProductID of 40.
* Lastly, we can run the Query of SELECT * FROM [Products] WHERE ProductID = 40 which gives us the final answer of :

### Boston Crab Meat 